### PR TITLE
Merge staging Locations changes

### PR DESCRIPTION
**Pull request by esoergel**
This has all been on staging for a while, and done QA/UAT there.  There's still some stuff we need to iterate on, so I can't merge the whole thing, so I've kept the feature preview in there, but now all it does is hide the locations menu items from the UI.

This adds in the restriction that a domain be standard plan or better to use locations, and a bunch of miscellaneous other changes.

http://manage.dimagi.com/default.asp?176034
@sravfeyn

### CODE REVIEW

File: corehq/apps/locations/views.py [hunk 1]
```diff
@@ -59,6 +59,10 @@ def dispatch(self, request, *args, **kwargs):
     def section_url(self):
         return reverse(LocationsListView.urlname, args=[self.domain])
 
+    @method_decorator(locations_access_required)
```
Comment (dimagimon) [inline corehq/apps/locations/views.py]: F811 redefinition of unused 'dispatch' from line 54

**Comment by esoergel** [comment]
(merge is blocked pending review of the domains in http://manage.dimagi.com/default.asp?176034)

File: corehq/apps/users/views/mobile/users.py [hunk 2]
```diff
@@ -159,10 +159,10 @@ def page_context(self):
             'data_fields_form': self.custom_data.form,
             'can_use_inbound_sms': domain_has_privilege(self.domain, privileges.INBOUND_SMS),
         }
-        if self.request.project.commtrack_enabled or self.request.project.locations_enabled:
+        if self.request.project.commtrack_enabled or self.domain_object.uses_locations:
```
Comment (sravfeyn) [inline corehq/apps/users/views/mobile/users.py]: Are `self.request.project` and `self.domain_object` point to same different, does this view has `domain_object` attribute (I couldn't find)?

File: corehq/apps/locations/management/commands/set_location_id.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def _is_location_domain(domain):
                 return domain_cache[domain]
             else:
                 domain_obj = Domain.get_by_name(domain)
-                val = domain_obj.commtrack_enabled or domain_obj.locations_enabled
+                val = domain_obj.commtrack_enabled or domain_obj.uses_locations
```
Comment (sravfeyn) [inline corehq/apps/locations/management/commands/set_location_id.py]: (not a biggy) Is it worth putting this into a domain method? Same condition appears again in the PR.

**Comment by sravfeyn** [comment]
:+1:

File: corehq/apps/locations/management/commands/set_location_id.py (hunk 3 above)
Comment (esoergel) [inline corehq/apps/locations/management/commands/set_location_id.py]: Actually `commtrack_enabled` is a prerequisite of `uses_locations`, so this is redundant.  I'll update.

File: corehq/apps/users/views/mobile/users.py (hunk 2 above)
Comment (esoergel) [inline corehq/apps/users/views/mobile/users.py]: They should point to the same thing.  `domain_object` is defined [here](https://github.com/dimagi/commcare-hq/blob/locations-for-robber-barons/corehq/apps/domain/views.py#L179-185), I think most reports have it available.

Comment (esoergel) [inline corehq/apps/users/views/mobile/users.py]: (I updated to at least be consistent in which one of them is used)

**Comment by esoergel** [comment]
Sheel finished review, this is g2m.

File: corehq/apps/reports/standard/sms.py [hunk 4]
```diff
@@ -297,7 +297,11 @@ def rows(self):
         data = SMSLog.by_domain_date(self.domain, startdate, enddate)
         result = []
 
-        reporting_locations_id = self.get_location_filter() if self.locations_enabled else []
+        direction_map = {
```
Comment (dimagimon) [inline corehq/apps/reports/standard/sms.py]: F841 local variable 'direction_map' is assigned to but never used

**Pull request merged**